### PR TITLE
Fix autocompletion after DOLLAR token

### LINKED ISSUE #53395: GDScript: `$` get node (dollar) operator completion issues
### Godot version

4.0.dev (865b62b)

### System information

Linux, Mageia 9 x86_64

### Issue description

While working on #53303, I found two issues with the completion for GDScript's `$` get equivalent operator (which is equivalent to calling the `get_node()` function). Both are still reproducible even once #53303 will be merged.

- [ ] Completion options are not limited to only valid NodePaths. All classes in the global scope seem included:

Screenshot_20211004_163122

- [ ] Completion a NodePath which needs quotes (e.g. with spaces or invalid characters for an identifier) will include two quotes at the end when pressing <kbd>Tab</kbd>:

Screenshot_20211004_163143

Screenshot_20211004_163243

### Steps to reproduce

- Open GDScript editor for a script attached to a node with child nodes
- Use the `$` operator and try to complete node paths, including paths with spaces

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by HolonProduction**
Fixes #53395

### CODE REVIEW

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -3127,7 +3130,7 @@ GDScriptParser::ExpressionNode *GDScriptParser::parse_get_node(ExpressionNode *p
 			path_state = PATH_STATE_SLASH;
 		}
 
-		make_completion_context(COMPLETION_GET_NODE, get_node, context_argument++);
+		make_completion_context(COMPLETION_GET_NODE, get_node, context_argument++, true);
```
Comment (Mickeon) [inline modules/gdscript/gdscript_parser.cpp:3133]: What does this `true` (`p_force` parameter) do here? Does the first line above not fix the issue on its own?

Comment (HolonProduction) [reply modules/gdscript/gdscript_parser.cpp:3133]: It is to ensure, that the first context gets overwritten, with the one which holds more information.
Currently this information (the Parser Node) is not used, as far as I know. But if someone wants to use it in the future it should get populated.
If we don't add the `true` we could delete the line entirely.

Comment (Mickeon) [reply modules/gdscript/gdscript_parser.cpp:3133]: Futureproofing in this case sounds good to me. I would suggest a comment but honestly the entire parser requires a few days to understand on its own, I don't think one would help sadly.

**Review by Mickeon — APPROVED** [review]
This _looks_ fine to me. Could you rebase this for good measure? Could use one final test.

**Comment by Mickeon** [comment]
I can confirm that this does in fact fix the issue and it does feel great.

But, as much as I dislike `%` for unique nodes, the PR does not address it for them since it is specific to the DOLLAR token.

bb4eb9bd4bac9483cd674406a8f60d4d

Although this is fine, if there were a more encompassing solution to the issue it would be much nicer.

**Comment by HolonProduction** [comment]
I actually considered fixing it for `%` as well. The problem is, that it could also be in a modulo context, where autocompletion popping up is far more annoying then the current behavior.

We could probably only make it pop up, if there is no other value directly in front of it. But that would require changes in other files as well. Might be worth to investigate for a future PR though.

**Comment by Mickeon** [comment]
<sub>Grrrr how I hate the `%`, should've only been `$%`</sub>

File: modules/gdscript/gdscript_parser.cpp (hunk 1 above)
Comment (HolonProduction) [reply modules/gdscript/gdscript_parser.cpp:3181]: Thinking about it again. I actually introduced a usage of the parser node in #85384 so this is important.

**Comment by HolonProduction** [comment]
Another problem about `%`: Autocompletion uses the parser node to decide which nodes are suggested. Without another character after the `%` we won't get a valid parser node and autocompletion will default to the behavior of `$` so the suggested nodes would be wrong.

I should probably create an issue for `%` and bundle the information there :thinking:

**Review by akien-mga — APPROVED** [review]
Looks good to me.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**